tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/OpenPaasConfiguration.java

```diff
@@ -1,6 +1,78 @@
 package com.linagora.tmail;
 
-import java.net.URL;
+import java.net.URI;
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Strings;
+
+public record OpenPaasConfiguration(Optional<AmqpUri> maybeRabbitMqUri, Optional<URI> openpaasApiUri, Optional<String> maybeAdminUser, Optional<String> maybeAdminPassword) {
```

Comment:
```suggestion
public record OpenPaasConfiguration(Optional<AmqpUri> maybeRabbitMqUri, URI openpaasApiUri, String maybeAdminUser, String maybeAdminPassword) {
```

Forgive me but this extension cannot work without OpenPaaS infos?

Reply:
Not it cannot, done.